report_thesis/src/sections/background.tex

```diff
@@ -205,6 +205,34 @@ \subsubsection{Support Vector Regression (SVR)}
 The optimization problem in \gls{svr} is to find the coefficients $\mathbf{w}$ and $b$ that minimize the norm of $\mathbf{w}$ (i.e., keep the regression function as flat as possible) while ensuring that most data points lie within the $\epsilon$ margin.
 
 \subsubsection{Extra Trees Regressor (ETR)}
+Before discussing the \gls{etr} model, it is important to first understand the concepts of decision trees and \gls{rf}.
+We give an overview of decision trees and \gls{rf} based on \citet{James2023AnIS}.
+Then, we introduce the \gls{etr} model based on \citet{geurtsERF}.
+
+A decision tree is a supervised learning model that partitions data into subsets based on feature values, resembling a tree structure.
+The goal is to create a tree that predicts the target variable by dividing the data into increasingly homogeneous subsets.
+Each internal node in the tree represents a decision based on a specific feature, while each leaf node represents a prediction for the target variable.
+The tree can make predictions for new data points by following a path from the root to a leaf node.
+
+\gls{rf} is an ensemble learning method that improves the accuracy and robustness of decision trees by building multiple trees and combining their predictions.
+Each tree in the forest is trained on a random subset of the data using bootstrap sampling, a method where samples are drawn with replacement from the original dataset.
+This technique allows each tree to be trained on a slightly different dataset, introducing variability and reducing overfitting.
+As a result, some data points may appear multiple times in the subset, while others may be omitted entirely.
+By training multiple trees on different subsets and then aggregating their predictions, the model achieves better generalization and robustness, leading to more accurate and reliable final predictions.
+
+Mathematically, the prediction of a \gls{rf} model can be represented as:
+
+$$
+f(x) = \frac{1}{M} \sum_{m=1}^{M} f_m(x),
+$$
+
+where $f_m(x)$ is the prediction of the $m$-th tree in the forest, and $M$ is the total number of trees.
+
+\gls{etr} is an extension of the \gls{rf} model that introduces additional randomness in the tree-building process, namely \textit{random feature selection} and \textit{random split points}.
+While \gls{rf} use bootstrap sampling to create diverse training sets, \gls{etr} goes further by selecting a random subset of features at each node and determining split points randomly within the range of those features.
```

Comment:
Is this right? I'm specifically referring to the `while` and `goes further` that imposes a certain relationship between the models. I'm not sure whether it's quote the right thing to highlight.

**From my understanding:**
In traditional Decision Tree algorithms, including those used in random forests, when deciding where to split a node, the algorithm evaluates several possible thresholds for each feature (or a randomly selected subset of features in the case of random forests).
It calculates some metric (like Gini impurity, information gain, or mean squared error) for each possible split and chooses the threshold that results in the best split according to that metric.
This process is computationally intensive because it involves evaluating many potential splits.

Extra-trees simplify and randomize this process.
Instead of searching for the optimal threshold for each feature, the algorithm for an extra-tree randomly selects a threshold for each feature under consideration at each split.
This means that the decision on where to split a node is made based on a randomly chosen value for each feature rather than a carefully calculated one.

I think you're trying to highlight how each model induces randomness in the process. I'm not sure Extra Trees also do bootstrap sampling. From what I understand, both uses a random subset of features at each node to determine split points. But then:
- Random Forests evaluate several possible thresholds for each feature in the selected subset and choose the optimal split based on a certain metric.
- Extra Trees introduce additional randomness by selecting the split points randomly within the range of the selected features instead of calculating the optimal thresholds.

So I don't think `goes further` is 100% right. Open to discussion if you have a different understanding!
I think it would be more appropriate to say that Extra Trees introduce randomness differently compared to Random Forests.

Reply:
```suggestion
While \gls{rf} use bootstrap sampling to create diverse training sets and select a random subset of features at each node to determine the best split based on various thresholds, \gls{etr} also selects a random subset of features at each node but introduces additional randomness by choosing split points randomly within the range of those features
```

Reply:
Agree